posts/mobs-are-not-always-destructive.md

```diff
@@ -0,0 +1,94 @@
+
+# There are... Guidelines
+
+Mobbing is meant to be an inclusive exercise so that no one individual on the team is always dictating what to do. Guidelines were created, therefore, to help enforce that mentality. They are as follows:
+
+1. Everyone focused on the goal (if you have to do something else step out)
+1. One person coding at a time
+1. Person coding switches every N minutes
+1. Problem solve as a group
+1. Person on the keyboard does not type anything they do not understand
+
+The whole group is centered around completing a single goal. Since only one individual is driving at a time at the keyboard, this allows for the entire team to participate, share, and collaborate on the group's solution.
+
+# How a Whole Mob can be Agile
+
+Worth highlighting that mob programming first was mentioned in "[Chapter 28: Mob Programming and the Transition to XP](https://www.worldcat.org/title/extreme-programming-perspectives/oclc/49663563)" of the _Extreme Programming Perspectives_ by Moses Hohman in 2003. However, just because it originated in the Extreme Programming (XP) methodology does not mean it cannot also be applied to other methodologies like scrum, kanban, scrum-ban, lean, etc. and here are two reasons why that is the case:
+
+1. At its core mobbing has only one purpose which is to deliver a single piece of work.
+1. A team does not jump into mobbing without a well-defined goal.
+
+Prior to the designated mobbing time, a shared goal is selected for the exercise which may push the team in the right direction and be utilized further on down the line. It can be easily incorporated into the project's agile methodology whether it is scrum, kanban, or XP. It is a way to work on a story, a task, or whatever the team deems as an integral step for everyone to continue going forward. This step will be as complex or simple as necessary because it depends on what the participating group is capable of completing.
+
+# Our Mobs and Recommendations
+
+## Team Capability and Capacity
+
+Devil's advocates will be quick to point out that mobbing directly reduces capacity across any team since all developers are involved in the same piece of work. While this is a valid concern from those concerned about velocity and throughput within a sprint, our most effective mobbing sessions were used for tackling large stories that probably would not have been completed within a sprint by a single developer anyway. These sessions were so successful for us that we even combined forces with our sister team to start in on new feature epics. For multiple teams that share a single backlog, this could be a preferable method of getting everyone on the same page as it cuts down on the number of design sessions that are necessary afterwards. All team members are now familiar enough with this shared codebase and each person has actually been hands-on with it. This means all the developers involved are going to be that much more productive in their work going forward.
```

Comment:
```suggestion
Devil's advocates will be quick to point out that mobbing directly reduces capacity across any team since all developers are involved in the same piece of work. While this is a valid concern from those concerned about velocity and throughput within a sprint, our most effective mobbing sessions were used for tackling large stories that probably would not have been completed within a sprint by a single developer anyway. These sessions were so successful for us that we even combined forces with our sister team to start on new feature epics. For multiple teams that share a single backlog, this could be a preferable method of getting everyone on the same page as it cuts down on the number of design sessions that are necessary afterwards. All team members are now familiar with this shared codebase and each person has actually been hands-on with it. This means all the developers involved are going to be that much more productive in their work going forward.
```